### ISSUE TITLE
Text Editor Theme not Applying to Notebooks

### ISSUE THREAD

**Issue by rctonnies** [jupyterlab/jupyterlab#5331]
Hi, I've not posted on here before; read through the guidelines and tried to find this issue in a previous thread.  

The Text Editor theme changes the style for everything except .ipynb lab files, which remain in the default Dark or Light styled theme.  Any other type of text file obeys the text editor theme selection on the fly and instantly.

Not sure if this is intentional or a problem just on my environment. Have tried fixing this for hours, with most forums referring to theme changes for Notebook, I've not found anything.  Would LOVE to be able to have the notebooks display the selected Editor theme.  :)

![ss1](https://user-images.githubusercontent.com/43343534/45629643-7a838f80-ba5c-11e8-8528-d5e24ba3cd20.png)
![ss2](https://user-images.githubusercontent.com/43343534/45629645-7b1c2600-ba5c-11e8-9280-ed457fb464a8.png)

**Comment by jasongrout** [comment]
The thinking here is that the "Text Editor theme" is referring to the text file editor tabs, not the generic code editor used also in the notebook. IIRC, right now there isn't a way to switch the theme for notebook code cell editors. Definitely +1 to making possible though - thanks for filing the issue!

**Comment by ian-r-rose** [comment]
Hi @rctonnies, for similar reasons to #3992/#1362, text editor keymaps are not applied to the notebook. It is a more specialized document than a plain text file, and it is difficult to combine arbitrary text editor themes with the rest of the document theme.

It is definitely possible to write an extension to do this if you want to explore it, however. I'd be happy to help guide in that process if anybody wants to take a crack at it.

**Comment by bollwyvl** [comment]
On Mon, Sep 17, 2018, 11:18 Ian Rose <notifications@github.com> wrote:

> Hi @rctonnies <https://github.com/rctonnies>, for similar reasons to #3992
> <https://github.com/jupyterlab/jupyterlab/issues/3992>/#1362
> <https://github.com/jupyterlab/jupyterlab/issues/1362>, text editor
> keymaps are not applied to the notebook.
>

Eek ... Don't want to think about keymaps!

> Itt is a more specialized document than a plain text file, and it is
> difficult to combine arbitrary text editor themes with the rest of the
> document theme.
>
> It is definitely possible to write an extension to do this if you want to
> explore it, however. I'd be happy to help guide in that process if anybody
> wants to take a crack at it.
>

Yum. Do want.

The base 'jupyter' theme, which is unapologetically python-focused, makes
some tokens very hard to distinguish for certain modes. Indeed, per
kernel/file type modes would be really nice.

While I don't like nested menus, since it's always picking from the list of
themes, perhaps having:

```
editor theme ->
[jupyter | ... | Zenburn] ->
[Editor | ... | Notebook] ->
[default | Ada | ... | Zsh]
```

Though it could be contextual, based on the active document, skipping a
layer.

Another thing I've been wondering about is a mime type of codemirror simple
modes...

>

**Comment by ian-r-rose** [comment]
Oops, meant to say "themes" in the first sentence above!

**Comment by Nathan-Furnal** [comment]
Any update on this issue?

I'd like to be able to use a dark theme because it's easier on my eyes with Zenburn for the text editor. Though there's the [V] next to Zenburn, jupyterlab doesn't change the layout.

On jupyter lab ``0.35.6`` and notebook ``5.7.8``.

**Comment by xgdgsc** [comment]
Could we combine https://github.com/dunovank/jupyter-themes to provide notebook themes? I also thought editor theme was for notebooks.

**Comment by udaychadha** [comment]
Checking in to see if there's any update on this. Just started using labs from notebooks and wanted to setup my notebook theme.

**Comment by damienstanton** [comment]
Even if this feature does not end up being implemented, it may simply be worth changing the verbiage of "Text Editor Theme". Many people (myself included) thought that this would mean the cells where text is edited, not the UI tabs.

Would it make sense to change this from "Text Editor Theme" to "UI Theme" or "Tab Theme"?

**Comment by kwcooper** [comment]
@xgdgsc I agree. I have had good milage with the updates with jupyter-lab support: https://github.com/dunovank/jupyterlab_darkside_ui
